src/ocl/convolutionocl.cpp

```diff
@@ -826,55 +827,98 @@ void ConvolutionDescriptor::GetSolutionsFallback(Handle& handle,
     ctx.SetStream(&handle);
     ctx.DetectRocm();
 
-    const auto wti2time = [](const float& wti) {
-        assert(wti != 0.0f);
-        if(wti <= 0.0f) // Return negative values as is, avoid DIV/0.
-            return wti;
-        return 10.0f / wti; // Assume WTI == 1.0 (100%) is 10 ms.
-    };
-
-    for(const auto& solver_id : solver::GetSolversByPrimitive(solver::Primitive::Convolution))
+    if(MIOPEN_ENABLE_HEUR && ConvHeur::IsApplicable(handle.GetDeviceName(), problem.conv_problem))
```

Comment:
@atamazov If the heuristic is not applicable for an arch/config the `else` branch would be taken and hence the wti machinery would kick in, please correct me if I am missing something!